src/client/datascience/jupyter/jupyterDebugger.ts

```diff
@@ -54,13 +53,18 @@ export class JupyterDebugger implements IJupyterDebugger, ICellHashListener {
         if (!notebook) {
             throw new Error('Notebook not initialized');
         }
-        const result = await this.installer.isProductVersionCompatible(
-            Product.ipykernel,
-            '>=6.0.0',
-            notebook.getKernelConnection()?.interpreter
-        );
+
+        let result = false;
+        const code = 'import ipykernel\nprint(ipykernel.__version__)';
+        const output = await kernel.executeHidden(code);
+
+        if (output[0].text) {
+            const majorVersion = Number(output[0].text.toString().charAt(0));
```

Comment:
There should be something already to parse semvers I think? Yeah it's an npm module for parsing them.